vm/src/org/mozartoz/truffle/translator/Translator.java

```diff
@@ -531,31 +535,35 @@ private Arity buildArity(OzArity arity) {
 	}
 
 	private OzNode translateBinaryOp(String operator, OzNode left, OzNode right) {
-		left = deref(left);
-		right = deref(right);
 		switch (operator) {
+		case "==":
+			return BuiltinsManager.createNodeFromFactory(EqualNodeFactory.getInstance(), left, right);
```

Comment:
The lines are a bit long, maybe we can simple have a helper function `createBinaryNode` or so.